transport/x/grpc/config_test.go

```diff
@@ -40,12 +40,24 @@ func TestNewTransportSpecOptions(t *testing.T) {
 	require.Equal(t, 2, len(transportSpec.OutboundOptions))
 }
 
+func TestConfigBuildInboundOtherTransport(t *testing.T) {
+	transportSpec := &transportSpec{}
+	_, err := transportSpec.buildInbound(&InboundConfig{}, testTransport{}, nil)
+	require.Equal(t, newTransportCastError(testTransport{}), err)
```

Comment:
In this case, I think it actually works, and in general I actually do want to test for the same error type - I think here this makes sense, but let's catch up on this offline later.